### PR TITLE
HAAR-872 remove group from user

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @DeleteMapping("/users/id/{userId}/groups/{group}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt:34]: I was unsure whether we needed /id/ as we're not providing a username endpoint anymore.
I ended up changing this to just /users/{userId}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        responseCode = "204",
+        description = "Deleted"
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Validation failed.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt:50]: Do we normally show this?  I can't remember

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          )
+        ]
+      )
+    ]
+  )
+  fun removeGroupByUserId(
+    @Parameter(description = "The userId of the user.", required = true) @PathVariable
+    userId: UUID,
+    @Parameter(description = "The group to be delete from the user.", required = true) @PathVariable
+    group: String
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt:80]: should description say group code - or is it group name?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt]: Updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt (hunk 3 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt]: Amended

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupController.kt]: Removed

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**